realm/realm-library/src/main/cpp/io_realm_internal_OsMap.cpp

```diff
@@ -56,22 +56,81 @@ JNIEXPORT jobject JNICALL
 Java_io_realm_internal_OsMap_nativeGetValue(JNIEnv* env, jclass, jlong map_ptr,
                                             jstring j_key) {
     try {
-        auto &dictionary = *reinterpret_cast<realm::object_store::Dictionary*>(map_ptr);
+        auto& dictionary = *reinterpret_cast<realm::object_store::Dictionary*>(map_ptr);
         JStringAccessor key(env, j_key);
-        JavaAccessorContext context(env);
-        return any_cast<jobject>(dictionary.get(context, StringData(key)));
+        const Optional<Mixed>& optional_result = dictionary.try_get_any(StringData(key));
```

Comment:
Could it be possible to use `auto get(Context&, StringData key) const;` instead of `try_get_any`. It will remove the following switch case as it handles the result boxing automatically.